File: payroll-api/src/main/java/com/company/payroll/model/StaffPromotion.java
package com.company.payroll.model;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
public class StaffPromotion {
    private Integer pId;
    private BigDecimal currentSalary;
    private BigDecimal promoteSalary;
    private LocalDate promoteDate;
    private String comment;
    private Title title;
    private StaffDetails staffDetails;
    private Integer titleNo;
    private Integer staffId;

//    private String comment;
//    private Integer eId;
//    private Integer mId;
//    private List<Title> title;
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/model/StaffPromotion.kt
package com.company.payroll.model

import com.company.payroll.annotation.NoArg
import java.math.BigDecimal
import java.time.LocalDate

@NoArg
data class StaffPromotion(
    var pId: Int,
    var currentSalary: BigDecimal,
    var promoteSalary: BigDecimal,
    var promoteDate: LocalDate,
    var comment: String,
    var title: Title?,
    var staffDetails: StaffDetails?,
    var titleNo: Int,
    var staffId: Int
)
